### PR TITLE
DOCSUP-5082: Docs for cache types

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation

### CODE REVIEW

File: docs/en/operations/caches.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- `uncompressed_cache` — Cache of uncompressed data used by table engines of the [MergeTree](../engines/table-engines/mergetree-family/mergetree.md) family.
+
+Additional cache types:
+- DNS cache
+- [regexp](../interfaces/formats.md#data-format-regexp) cache
+- compiled expressions cache
+- [Avro format](../interfaces/formats.md#data-format-avro) schemas cache
+- [dictionaries data cache](../sql-reference/dictionaries/index.md)
+
+Not directly used:
```
Comment (alexey-milovidov) [inline docs/en/operations/caches.md:21]: Not directly used -> Indirectly used

File: docs/en/operations/caches.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- page cache OS
```
Comment (alexey-milovidov) [inline docs/en/operations/caches.md:22]: OS page cache

Comment (olgarev) [reply docs/en/operations/caches.md]: @alexey-milovidov Fixed that, thanks.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Do we have links to the SYSTEM DROP ... CACHE commands easily reachable from this page?

**Comment by olgarev** [comment]
> Do we have links to the SYSTEM DROP ... CACHE commands easily reachable from this page?

@alexey-milovidov Here https://github.com/ClickHouse/ClickHouse/pull/19408

**Pull request merged**